README.md

```diff
@@ -153,35 +153,48 @@ interface MutatorService
 
 ![Mutator class](images/Mutator-s5.png)
 
-### GenePool
+### GenePoolService
 ```
-class GenePool
+class GenePoolService
     private final RandomProvider randomProvider
     private final MutatorService mutator
     private final Evaluator evaluator
     private final CrossoverHandler crossoverHandler
     private final List<Gene> poolOfGenes
-    private int generation
     
-    // getters and setters
     List<Gene> initializeGenes()
     void makeMutation()
     void evaluateFitness()
-    void performEvolution()
     void makeCross()
+    void getPoolOfGenes()
 ```
-<p>GenePool class is responsible for gene evolution by evoking methods in proper order. GenePool is also a container for genes.</p>
-<p>List of Genes is initialized with class creation with int argument given to constructor, using method generateGenes(int number).</p>
+
+<p>GenePoolService class carry all logic responsible for gene evolution by evoking methods in proper order.GenePoolService is also a container for genes.</p>
```

Comment:
I think GenePoolService should not be container for genes, because then we will mix logic and data again.